### PR TITLE
Ag 13351 faq component refactor

### PR DESCRIPTION
**Pull request by markdurrant**
Refactor homepage FAQ component. 

- Minor design tweaks (hover/cursor/ect)
- Hardcoded links 
- Added to website shared 
- Animate expanding sections 
- Move `<Collapsible />` component to website shared

### CODE REVIEW

File: documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[
+    {
+        "question": "What is the difference between AG Grid Community and Enterprise?",
+        "answer": "<a href='/react-data-grid/getting-started/' target='_blank' rel='noopener noreferrer'>AG Grid Community</a> is the free version with basic features, while <a href='/license-pricing/' target='_blank' rel='noopener noreferrer'>AG Grid Enterprise</a> offers additional advanced features and support."
```
Comment (taktran) [inline documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json]: The links aren't going to work on archive pages. We'll nee to rethink it. Maybe use the markdoc transform approach we talked about.

Comment (markdurrant) [reply documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json]: Now using Markdoc as suggested.

File: documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        "question": "What is the difference between AG Grid Community and Enterprise?",
+        "answer": "[AG Grid Community](./getting-started/) is the free version with basic features, while [AG Grid Enterprise](/license-pricing/) offers additional advanced features and support."
+    },
+    {
+        "question": "Can I create a React Table for free?",
+        "answer": "Yes, you can create a React Table for free using our open-source library, [AG Grid Community](./getting-started/)."
+    },
+    {
+        "question": "Will my React Tables be compatible with AG Charts?",
+        "answer": "Yes, AG Grid is designed to be compatible with AG Charts, allowing seamless integration between React Tables and [React Charts](/charts/react/quick-start/)."
```
Comment (taktran) [inline documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json]: This link isn't going to work on staging. To make it work, we might need to add a charts website identifier eg,

```suggestion
        "answer": "Yes, AG Grid is designed to be compatible with AG Charts, allowing seamless integration between React Tables and [React Charts](%charts%/react/quick-start/)."
```

Maybe for another PR though

**Review by taktran — APPROVED** [review]
Looks good, just an open q about making the links work for charts website

File: documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json (hunk 2 above)
Comment (markdurrant) [reply documentation/ag-grid-docs/src/content/homepage/homepageFAQs.json]: For now making the charts link point to prod solves issue

**Pull request merged**